pkg/hub/controllers/certrotation_controller.go

```diff
@@ -65,7 +73,7 @@ func NewCertRotationController(
 				Namespace:     namespace,
 				Name:          clusterProxyAddOnSecet,
 				Validity:      TargetCertValidity,
-				HostNames:     []string{fmt.Sprintf("%s.%s.svc", "route", namespace)}, //TODO: using an actual service
+				HostNames:     []string{fmt.Sprintf("%s.%s.svc", "route", namespace), serviceIP},
```

Comment:
a dns name or an IP?

Reply:
It should be clusterIP since route's policy is passthrough